src/to-govspeak.js

```diff
@@ -114,6 +114,54 @@ service.addRule('cleanUpNestedLinks', {
   }
 })
 
+// Google docs has a habit of producing nested lists that are not nested
+// with valid HTML. Rather than embedding sub lists inside an <li> element they
+// are nested in the <ul> or <ol> element.
+service.addRule('invalidNestedLists', {
+  filter: (node) => {
+    const nodeName = node.nodeName.toLowerCase()
+    if ((nodeName === 'ul' || nodeName === 'ol') && node.previousElementSibling) {
+      const previousNodeName = node.previousElementSibling.nodeName.toLowerCase()
+      return previousNodeName === 'li'
+    }
+  },
+  replacement: (content, node) => {
+    content = content
+      .replace(/^\n+/, '') // remove leading newlines
+      .replace(/\n+$/, '') // replace trailing newlines
+      .replace(/\n/gm, '\n    ') // indent
+
+    return '    ' + content + '\n'
+  }
+})
+
+// This is ported from https://github.com/domchristie/turndown/blob/80297cebeae4b35c8d299b1741b383c74eddc7c1/src/commonmark-rules.js#L61-L80
+// It is modified to handle items from the invalidNestedLists as if these
+// are ordered lists they will be output with incorrect numbering.
+service.addRule('listItems', {
+  filter: 'li',
+  replacement: function (content, node, options) {
+    content = content
+      .replace(/^\n+/, '') // remove leading newlines
+      .replace(/\n+$/, '\n') // replace trailing newlines with just a single one
+      .replace(/\n/gm, '\n    ') // indent
+
+    let prefix = options.bulletListMarker + '   '
+    const parent = node.parentNode
+    if (parent.nodeName.toLowerCase() === 'ol') {
+      const start = parent.getAttribute('start')
+      const listItems = Array.prototype.filter.call(
+        parent.children, (element) => element.nodeName.toLowerCase() === 'li'
+      )
+      const index = Array.prototype.indexOf.call(listItems, node)
+      prefix = (start ? Number(start) + index : index + 1) + '.  '
```

Comment:
I'm afraid the interaction between 'start' and 'index' is confusing me, but it does look like there's more than one case to test here. Would it be possible to break this up in some way, to make it more transparent?

Reply:
I don't really want to change the code here as I want to maintain it being as close to the ported code as possible, but happy to add a test for the start condition. 2 mins.

Reply:
Actually just tried out the start code and it does nothing for us when the markdown is converted back to govspeak so will just remove it.

Example:

HTML
```html
<ol start=5>
  <li>item 1</li>
  <li>item 1</li>
</ol>
```

Converts to govspeak/md:

```
5.  item
6.  item
```

Then from govspeak to HTML:

```html
<div class="gem-c-govspeak govuk-govspeak ">
  <ol>
     <li>item</li>
     <li>item</li>
   </ol>
</div>
```

Reply:
I've updated now